Rooms 2.0 documentation

soapdog opened the pull request:
This PR adds a section for Rooms 2.0 documentation. We can see an online version of this document in [my fork](https://soapdog.github.io/scuttlebutt-protocol-guide/#rooms).

A lot of the new SSB technologies have written specifications already, so to avoid duplicating their content, these new sections in the protocol guide will provide an _at a glance_ information about them and link to the specification documents.

mycognosist commented:
@soapdog 

Great work on this!

A few things I noticed:

1) The solid line and dotted line in the tunnel connection SVG label are very difficult to distinguish. At first I was thought you had accidentally repeated the same label twice. Perhaps you could have Alice connecting to the Room in the first label (direct connection), with a green dot, green A, green solid line and then pink R dot (Alice -> Room). Then the second label (indirect connection) could be a blue dot, blue B, blue dotted line and then green A and dot (Bob ---> Alice). 

2) I wonder if it might be an idea to explicitly mention that the "peer" in the tunnel section is "the room"? Maybe something like this:

> A tunnel is an indirect connection between two Scuttlebutt peers which allows them to communicate as if they were on the same local network. This tunnel is made possible by the Room server which has a stable and publicly available Internet connection. The Room acts as a connection facilitator by wrapping the indirect connection between the other peers to create the tunnel.

3) One small typo I noticed: "Once the user navigates of that URI" should be "Once the user navigates **to** that URI"

I can make a few more minor grammatical changes if you're keen, but they're not game-changers. Again, nice work on this!

soapdog commented:
>     1. The solid line and dotted line in the tunnel connection SVG label are very difficult to distinguish. At first I was thought you had accidentally repeated the same label twice. Perhaps you could have Alice connecting to the Room in the first label (direct connection), with a green dot, green A, green solid line and then pink R dot (Alice -> Room). Then the second label (indirect connection) could be a blue dot, blue B, blue dotted line and then green A and dot (Bob ---> Alice).

I have pushed a new commit with a reworked graph. I've changed the color for the dashed line and made the dashes more visible. Please, check if it is better, I'm keen on feedback about it.
> 
>     2. I wonder if it might be an idea to explicitly mention that the "peer" in the tunnel section is "the room"? Maybe something like this:
> 
> 
> > A tunnel is an indirect connection between two Scuttlebutt peers which allows them to communicate as if they were on the same local network. This tunnel is made possible by the Room server which has a stable and publicly available Internet connection. The Room acts as a connection facilitator by wrapping the indirect connection between the other peers to create the tunnel.
> 

I stole that text wholesale. It is now on the docs. Thanks for it.

>     1. One small typo I noticed: "Once the user navigates of that URI" should be "Once the user navigates **to** that URI"
> 
> 

Fixed.

> I can suggest a few more minor grammatical changes if you're keen, but they're not game-changers. Again, nice work on this!

Please do! :D
